verify-cborld-presentation/src/pages/result.tsx

```diff
@@ -0,0 +1,153 @@
+/*
+ * Copyright 2019 - MATTR Limited
+ * All rights reserved
+ * Confidential and proprietary
+ */
+
+import { Box, Link, StylesProvider } from "@material-ui/core";
+import { encode as base64Encode, decode as base64Decode } from "@stablelib/base64";
+import base32Decode from "base32-decode";
+import Lottie from "lottie-react";
+import { inflate } from "pako";
+import React, { useEffect, useState } from "react";
+import styled from "styled-components";
+
+import errorAnimation from "../assets/lotti/error.json";
+import loadingSpinnerAnimation from "../assets/lotti/loadingSpinner.json";
+import successAnimation from "../assets/lotti/success.json";
+import { Credential, GenericButton } from "../components";
+import type { Credential as CredentialType } from "../service/types";
+import { isCredential } from "../service/types/validation/credential";
+
+type TrustedIssuer = {
+  did: string;
+  domain: string;
+};
+const trustedIssuers: TrustedIssuer[] = [
+  // example issuer
+  // {
+  //   did: "did:key:z6Mkne6kfQ2bY396dKTuraGLxiDoBrXXjVvH2Zv7nA1MnDoM",
+  //   domain: "localhost:3000",
+  // },
+];
+export interface ResultPageProps {
+  qrCode: string | undefined;
+}
+
+type VerifiationStatus = "checking" | "failed" | "success";
+
+export const ResultPage: React.FC<ResultPageProps> = (props: ResultPageProps) => {
+  const [credential, setCredential] = useState<CredentialType>();
+  const [trustedIssuer, setTrustedIssuer] = useState<string>();
+  const [verificationState, setVerificationState] = useState<VerifiationStatus>("checking");
+
+  useEffect(() => {
+    const qrCode = props.qrCode;
+    if (qrCode) {
```

Comment:
Maybe add an `isLoading` state or similar, otherwise, the webapp will hit the platform API (`/core/v1/linkeddata/convert`) on every rerender. 😄

Reply:
Hmm, maybe not, since this callback function will only be invoked when `props.qrCode` is changed.